updater/lib/dependabot/job.rb

```diff
@@ -43,124 +45,180 @@ class Job
+                                              T::Array[T::Hash[String, T.any(T::Boolean, String)]])
+      @dependencies                   = T.let(attributes.fetch(:dependencies), T.nilable(T::Array[T.untyped]))
+      @existing_pull_requests         = T.let(attributes.fetch(:existing_pull_requests), T::Array[T.untyped])
       # TODO: Make this hash required
       #
       # We will need to do a pass updating the CLI and smoke tests before this is possible,
       # so let's consider it optional for now. If we get a nil value, let's force it to be
       # an array.
-      @existing_group_pull_requests   = attributes.fetch(:existing_group_pull_requests, []) || []
-      @experiments                    = attributes.fetch(:experiments, {})
-      @ignore_conditions              = attributes.fetch(:ignore_conditions)
-      @package_manager                = attributes.fetch(:package_manager)
-      @reject_external_code           = attributes.fetch(:reject_external_code, false)
-      @repo_contents_path             = attributes.fetch(:repo_contents_path, nil)
-
-      @requirements_update_strategy   = build_update_strategy(
-        **attributes.slice(:requirements_update_strategy, :lockfile_only)
-      )
-
-      @security_advisories            = attributes.fetch(:security_advisories)
-      @security_updates_only          = attributes.fetch(:security_updates_only)
-      @source                         = build_source(attributes.fetch(:source))
-      @token                          = attributes.fetch(:token, nil)
-      @update_subdependencies         = attributes.fetch(:update_subdependencies)
-      @updating_a_pull_request        = attributes.fetch(:updating_a_pull_request)
-      @vendor_dependencies            = attributes.fetch(:vendor_dependencies, false)
+      @existing_group_pull_requests   =  T.let(attributes.fetch(:existing_group_pull_requests, []) || [],
+                                               T::Array[T.untyped])
+      @experiments                    =  T.let(attributes.fetch(:experiments, {}),
+                                               T.nilable(T::Hash[T.untyped, T.untyped]))
+      @ignore_conditions              =  T.let(attributes.fetch(:ignore_conditions), T::Array[T.untyped])
+      @package_manager                =  T.let(attributes.fetch(:package_manager), String)
+      @reject_external_code           =  T.let(attributes.fetch(:reject_external_code, false), T::Boolean)
+      @repo_contents_path             =  T.let(attributes.fetch(:repo_contents_path, nil), T.nilable(String))
+
+      @requirements_update_strategy   = T.let(build_update_strategy(
+                                                **attributes.slice(:requirements_update_strategy, :lockfile_only)
+                                              ), T.nilable(String))
+
+      @security_advisories            = T.let(attributes.fetch(:security_advisories), T::Array[T.untyped])
```

Comment:
Well... for example security_advisories is something like `T::Array[T::Hash[String,T.any(String, T::Array[String]])` which is kind of difficult for me to parse...

It would bring a lot of clarity to construct the actual `Dependabot::SecurityAdvisory` in the initializer and present it as `security_advisories` (`T::Array[Dependabot::SecurityAdvisory]`) instead of presenting the raw data and only using `Dependabot::SecurityAdvisory` when calling `security_advisories_for`.

The others are kind of similar, we have proper classes that have types we could use but we'd need to construct it here and then find all the places that currently use the array of hashes, so I am defering that to a future PR.